extract.py

```diff
@@ -0,0 +1,81 @@
+"""Script that connects to 'data-eng-plants-api'
+   Extracts the data
+   Puts it into a Pandas DataFrame"""
+import logging
+from multiprocessing import Pool, cpu_count
+import requests
+import pandas as pd
+
+
+URL = "https://data-eng-plants-api.herokuapp.com/plants/"
+TOTAL_NUMBER_OF_PLANTS = 50
+
+
+def get_plant_data(plant_id: int) -> dict:
+    """GET request to collect plant data from API."""
+    try:
+        response = requests.get(URL + str(plant_id), timeout=1000)
+        if response.status_code == 200:
+            return response.json()
+        return {"error": 404, "message": f"Failed to retrieve data for plant ID {plant_id}."}
+    except requests.RequestException as e:
+        logging.error("Request failed for plant ID %d: %s", plant_id, e)
+        return {"error": 500, "message": f"Server error for plant ID {plant_id}."}
+
+
+def extract_plant_data(response: dict) -> dict:
+    """Extracts plant data from the API response and handles missing fields."""
+    botanist_name = response.get("botanist", {}).get("name", "").split()
+    return {
+        "botanist_first_name": botanist_name[0] if len(botanist_name) > 0 else None,
+        "botanist_last_name": botanist_name[1] if len(botanist_name) > 1 else None,
+        "botanist_email": response.get("botanist", {}).get("email"),
+        "botanist_phone_number": response.get("botanist", {}).get("phone"),
+        "plant_name": response.get("name"),
```

Comment:
Did we conclude that there is always at least a 'name' value? (not accounting for Nones like with scientific name)